client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +49,65 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final boolean subscribedForRetention;
+    private final boolean autoTruncateAtLastCheckpoint;
+
+    public enum ReaderGroupRetentionConfig {
+        /**
+         * This {@link ReaderGroup} is not a subscriber and does not participate in Consumption Based Retention.
+         *
+         * See <a href="https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention">Consumption Based Retention</a>
+         *
+         */
+        NONE(false, false),
+
+        /**
+         * This {@link ReaderGroup} is a subscriber and can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream according to its read positions. These read positions have to be provided
+         * manually in the form of a {@link StreamCut}, see this method:
+         *
+         * {@link ReaderGroup#updateRetentionStreamCut(Stream, StreamCut)}
+         *
+         * See <a href="https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention">Consumption Based Retention</a>
+         *
+         */
+        TRUNCATE_AT_USER_STREAMCUT(true, false),
+
+        /**
+         * This {@link ReaderGroup} is a subscriber and can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream using the read positions from the last {@link Checkpoint}.
```

Comment:
```suggestion
         * This {@link ReaderGroup} requests that data be retained through the Readers' most recent checkpoint.
         * This is subject to the limits of the {@link StreamConfiguration.RetentionPolicy}. 
```

Reply:
Changed the javadoc keeping this in mind.